robot_smach_states/src/robot_smach_states/navigation/navigate_to_observe.py

```diff
@@ -71,8 +73,11 @@ def generateConstraint(self):
                 pci = pci + "-(x-%f)*%f+(y-%f)*%f > 0.0 " % (xs, dy, ys, dx)
 
         else:  # If not, simply use the x and y position
-            ro = "(x-%f)^2+(y-%f)^2 < %f^2" % (x, y, self.radius+0.075)
-            ri = "(x-%f)^2+(y-%f)^2 > %f^2" % (x, y, self.radius-0.075)
+            outer_radius = max(0., self.radius + self.margin)
+            inner_radius = max(0., self.radius - self.margin)
+
```

Comment:
I dont think that is necessary. inner_radius may be 0. Outer radius is at least equal to the margin. If we assume a competent input this will not go wrong.

Reply:
I think you are right in the fact that it is not necessary. The outer radius does not have to be equal to the margin in case of incompetent input (which I think we should take into consideration to the fact that you should get an understandable error). In the case of a negative radius the outer_radius can become zero, however this is caught down the pipeline in a clear enough (at least in my opinion) warning (in the getPlan function of the NavigateTo class)